### PR TITLE
Add tests for CRP and Math Utilties

### PR DESCRIPTION
**Pull request by srvasude**
Update logsumexp to handle infs and -inf values (and be a little more numerically stable).

### CODE REVIEW

File: cxx/Makefile [hunk 1]
```diff
@@ -37,8 +37,8 @@ LTEST		= $(wildcard $(TEST_DIR)/*.cc)
 %.out : %.cc $(LOBJECT)
 	$(CXX) $(CXXFLAGS) $(INCFLAGS) -o $@ $(LOBJECT) $<
 
-.PHONY: tests
-tests: $(LTEST:.cc=.out) hirm.out
+.PHONY: integration_tests
```
Comment (ThomasColthurst) [inline cxx/Makefile:40]: Make sure to also update the README with the new command for running the integration tests.

File: cxx/hirm_test.cc [hunk 2]
```diff
@@ -0,0 +1,45 @@
+// Apache License, Version 2.0, refer to LICENSE.txt
+
+#define BOOST_TEST_MODULE test BetaBernoulli
```
Comment (ThomasColthurst) [inline cxx/hirm_test.cc:3]: Update test name.

File: cxx/util_math_test.cc [hunk 3]
```diff
@@ -0,0 +1,124 @@
+// Apache License, Version 2.0, refer to LICENSE.txt
+
+#define BOOST_TEST_MODULE test BetaBernoulli
```
Comment (ThomasColthurst) [inline cxx/util_math_test.cc:3]: And here.

File: cxx/Makefile (hunk 1 above)
Comment (srvasude) [reply cxx/Makefile:40]: Done.

File: cxx/hirm_test.cc [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+  weights = crp.tables_weights();
+  BOOST_CHECK_CLOSE(weights[0], 2., 1e-6);
+  BOOST_CHECK_CLOSE(weights[1], 1., 1e-6);
+  BOOST_CHECK_CLOSE(weights[2], 1., 1e-6);
+  BOOST_CHECK_CLOSE(weights[3], alpha, 1e-6);
+
+  BOOST_CHECK_CLOSE(crp.logp(0), log(2. / (4. + alpha)), 1e-6);
+  BOOST_CHECK_CLOSE(crp.logp(1), log(1. / (4. + alpha)), 1e-6);
+  BOOST_CHECK_CLOSE(crp.logp(2), log(1. / (4. + alpha)), 1e-6);
+}
```
Comment (emilyfertig) [inline cxx/hirm_test.cc:45]: Add tests that call `sample` and `logp_score` too?

Comment (srvasude) [reply cxx/hirm_test.cc:103]: Done.

**Review by emilyfertig — APPROVED** [review]

**Pull request merged**